daemon/api.go

```diff
@@ -1073,6 +1076,29 @@ func getChanges(c *Command, r *http.Request, user *auth.UserState) Response {
 		return BadRequest("select should be one of: all,in-progress,ready")
 	}
 
+	if wantedName := query.Get("for"); wantedName != "" {
+		selectFilter := filter
+		filter = func(chg *state.Change) bool {
+			if !selectFilter(chg) {
+				return false
+			}
+
+			var snapNames []string
+			if err := chg.Get("snap-names", &snapNames); err != nil {
+				logger.Noticef("cannot get snap-name for change %v", chg.ID())
+				return false
+			}
+
+			for _, snapName := range snapNames {
```

Comment:
Looking at this I wonder if we should have "utils.Contains(needle interface{}, haystack {}interface)" for this (and similar constructs). Loosing type safety of course :/